src/jit/lower.cpp

```diff
@@ -3953,6 +4025,16 @@ void Lowering::DoPhase()
         currBlock = block;
         comp->compCurBB = block;
 
+#if !defined(_TARGET_64BIT_)
+        // Decompose TYP_LONG nodes
+        for (stmt = block->bbTreeList; stmt != nullptr; stmt = stmt->gtNext)
+        {
+            comp->compCurStmt = stmt;
+            comp->fgWalkTreePost(&stmt->gtStmt.gtStmtExpr, &Lowering::DecompNodeHelper, this, true);
+            stmt = comp->compCurStmt;
+        }
+#endif
+
         /* Walk the statement trees in this basic block */
```

Comment:
Is the reason for pulling this out into a separate loop because the decomposition might turn a single statement into two?